Grant the project_manager `publicize_image`

squaremo opened the pull request:
To be able to save server snapshots and make them available to UNI users*, the service accounts used for UNI need to have publicize_image. Here I've simply followed what compute.py does, using the same predicate (`is_project_manager`) and the permission `publicize_image` as mentioned in

    https://docs.openstack.org/glance/latest/_modules/glance/policies/image.html

*Snapshots are created private, and can only become available across UNI projects by being made public.

From https://docs.openstack.org/ocata/config-reference/image/policy.json.html I gather than it's only publicize_image that's restricted to admin (and needs to be changed here).

squaremo committed:
Grant the project_manager `publicize_image`

To be able to save server snapshots and make them available to UNI
users, the service accounts used for UNI need to have
publicize_image. Here I've simply followed what compute.py does, using
the name `publicize_image` as mentioned in

    https://docs.openstack.org/glance/latest/_modules/glance/policies/image.html

squaremo commented on pyproject.toml:
```diff
@@ -30,6 +30,7 @@ dependencies = [
 	"cinder ~= 25.0",
 	"neutron ~= 25.0",
 	"nova ~= 30.0",
+    "glance ~= 29.0",
```
Version of Glance in Dalmation.

spjmurray commented on unikorn_openstack_policy/image.py:
```diff
@@ -0,0 +1,53 @@
+from oslo_policy import policy
+from unikorn_openstack_policy import base
+
+PUBLICIZE = 'publicize_image'
```
Looks accurate, as much as groking openstack can every be

spjmurray reviewed (approved):
I believe this looks good, what does the output from the generate command look like?

squaremo committed:
Add image policy entry points to project file

yankcrime reviewed (approved):
Might as well update the README with the example while you're here.

squaremo committed:
Mention image service policy change in README.md

gmoss-nscale commented:
The doc you are working from is an old version And the request does not make sense and publicize_image is used to make images in glance public to the whole of openstack not just that project.

ref: https://docs.openstack.org/glance/2023.2/configuration/glance_policy.html
```
publicize_image
Default:
rule:context_is_admin

Operations:
PATCH /v2/images/{image_id}

Scope Types:
project
```

gmoss-nscale commented:
one ask if you can adjust to include the existing rule as we have another role called `image-publisher` that is used to create the nks images.

ref: https://github.com/nscaledev/openstack-deployment/blob/main/config/glance/policy.yaml
```
"publicize_image": "role:admin or role:image-publisher"
``` 
if you can then send me the updated generated policy I can look to get this in.

squaremo committed:
Account for existing configuration

This may be used to overwrite configuration; therefore, include the
role image-publisher in the rule, so it doesn't accidentally get
excluded. (Another role, admin, is implicitly incuded because it's the
default.)

spjmurray reviewed (approved):
Tested in a development cluster and it does work... with some tweaks.

squaremo commented:
> Tested in a development cluster and it does work... with some tweaks.

Tweaks needed for your specific cluster, or in general?
